Clarify how to use variables with handlers (#54928)
ansible/ansible: docs/docsite/rst/user_guide/playbooks_intro.rst

            name: apache
            state: restarted

> Avoid using variables in handler names. Since handler names are templated early on, all variables may not be available, resulting in play failure. It is possible to use variables in handler task parameters. For example, if the name of a service varies slightly by distribution, the service name can be set using ``include_vars`` (or any place where variables can be set).

Reviewer: I wonder if we should clarify that in case users choose to use variables in handler names then changing that variable does not result in some kind of magic dynamic creation of new handlers, which people thought is the case in that issue it seemed.

Reviewer 2: diff name == diff handler, so that would be a specific feature of doing this

Here is an example using ``include_vars``:

  .. code-block:: yaml+jinja

    tasks:
      - name: Set variables based on distribution

>         include_vars: "{{ ansible_facts.distributon }}.yml"

Reviewer 3: ```suggestion
        include_vars: "{{ ansible_facts.distribution }}.yml"
```

Author: Thank you for catching this!

>         include_vars: "{{ ansible_facts.distribution }}.yml"

Reviewer 2: include vars are 'host vars'

            name: apache
            state: restarted

> Avoid using variables in handler names. Since handler names are templated early on, all variables may not be available, resulting in play failure. If a handler name does contain a variable, changing that variables mid-play **will not** result in newly created handler.

Reviewer 4: ```suggestion
You may want your Ansible handlers to use variables. For example, if the name of a service varies slightly by distribution, you want your output to show the exact name of the restarted service for each target machine. Avoid placing variables in the name of the handler. Since handler names are templated early on, Ansible may not have a value available for a handler name like this::

   handlers:
   # this handler name may cause your play to fail!
   - name: restart "{{ web_service_name }}"

If the variable used in the handler name is not available, the entire play fails. Changing that variable mid-play **will not** result in newly created handler.
```

> It is possible to use variables in handler *task parameters*. For example, if the name of a service varies slightly by distribution, the service name can be set using ``include_vars``.

Reviewer 4: ```suggestion
Instead, place variables in the task parameters of your handler. You can load the values using ``include_vars`` like this:
```

> Here is an example using ``include_vars``:

Reviewer 4: ```suggestion

```

> 

Reviewer 4: ```suggestion
```

            name: apache
            state: restarted

You may want your Ansible handlers to use variables. For example, if the name of a service varies slightly by distribution, you want your output to show the exact name of the restarted service for each target machine. Avoid placing variables in the name of the handler. Since handler names are templated early on, Ansible may not have a value available for a handler name like this::

    handlers:

>     # this handler name may cause your play to fail!

Reviewer 4: ```suggestion
      # this handler name may cause your play to fail!
```

>     - name: restart "{{ web_service_name }}"

Reviewer 4: ```suggestion
      - name: restart "{{ web_service_name }}"
```
